internal/common/utils/pods.go

```diff
@@ -115,7 +115,11 @@ func ListWithPredicate(nodeLister corelisters.NodeLister, predicate NodeConditio
 
 	var filtered []*v1.Node
 	for i := range nodes {
-		if predicate(nodes[i]) {
+		matches, err := predicate(nodes[i])
+		if err != nil {
+			return nil, err
+		}
+		if matches {
```

Comment:
This change is needed because we don't want to ignore errors caused by:
`PodMatchesNodeSelectorAndAffinityTerms ` -> `v1affinityhelper.GetRequiredNodeAffinity(driver).Match(node)`